server/neptune/gateway/api/controller.go

```diff
@@ -0,0 +1,46 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+// Controller is a simple generic controller that converts a request and hands it off
+// providing a level of consistency across API handling.
+type Controller[Request any] struct {
```

Comment:
Im not clear on the abstraction here, could you explain it again? I guess the word `Controller` throws me off.

Reply:
It's just a convention in atlantis to use controllers as the entrypoint for external requests, probably stems from MVC. For example: https://github.com/lyft/atlantis/blob/4fed9a2b48efd7854850b421153265a11cec9fba/server/lyft/gateway/events_controller.go#L151-L152